contracts/stellar-interchain-token-service/src/token_metadata.rs

```diff
@@ -53,9 +53,18 @@ pub fn token_metadata(
     native_token_address: &Address,
 ) -> Result<TokenMetadata, ContractError> {
     let token = token::Client::new(env, token_address);
-    let decimals = token.decimals();
-    let name = token.name();
-    let symbol = token.symbol();
+    let decimals = token
+        .try_decimals()
+        .map_err(|_| ContractError::InvalidTokenAddress)?
+        .map_err(|_| ContractError::InvalidTokenAddress)?;
```

Comment:
this is a bit confusing, take a look at what the two nested results correspond to. they fail due to different errors. We might still want to always return this error in both cases, but would be good to understand the difference